hf_hub.py

```diff
@@ -18,100 +18,111 @@
 import torch
 
 from hubconf import get_prefix_state_dict
-from radio.hf_model import ERADIOConfig, ERADIOModel, RADIOConfig, RADIOModel
-
-
-hugginface_repo_mapping = {"RADIO": "nvidia/RADIO", "E-RADIO": "nvidia/E-RADIO"}
+from radio.hf_model import RADIOConfig, RADIOModel
 
 
 def main():
     """Main Routine.
 
-    Push a RADIO model to Hugging.
+    Construct and optionall push a RADIO model to Hugging.
 
     Usage:
 
-    python3 -m hf_hub --model RADIO --checkpoint-path /path/to/checkpoint.pth.tar
+    python3 -m hf_hub --model <model-name> --checkpoint-path <checkpoint-path> [--push]
+
+    Examples:
+
+    python3 -m hf_hub --hf-repo nvidia/RADIO --checkpoint-path radio_v2.1_bf16.pth.tar --version radio_v2.1 --push
+    python3 -m hf_hub --hf-repo nvidia/E-RADIO --checkpoint-path eradio_v2.pth.tar --version e-radio_v2
+
     """
     parser = argparse.ArgumentParser()
     parser.add_argument(
         "--checkpoint-path", help="Path to the pretrained weights", required=True
     )
+    parser.add_argument("--hf-repo", help="Path to the HuggingFace repo", required=True)
     parser.add_argument(
-        "--model", help="RADIO model type", required=True, choices=["RADIO", "E-RADIO"]
+        "--torchhub-repo", help="Path to the TorchHub repo", default="NVlabs/RADIO"
     )
+    parser.add_argument("--version", help="(E-)RADIO model version", required=True)
     parser.add_argument(
         "--huggingface-repo-branch",
         help="HuggingFace repository branch to push to",
         default="main",
     )
+    parser.add_argument(
+        "--push", help="Push the model to HuggingFace", action="store_true"
+    )
     args = parser.parse_args()
 
     # Load the checkpoint and create the model.
     checkpoint = torch.load(args.checkpoint_path, map_location="cpu")
     model_args = checkpoint["args"]
 
     # Extract the state dict from the checkpoint.
-    state_dict = checkpoint["state_dict"]
+    if "state_dict_ema" in checkpoint:
+        state_dict = checkpoint["state_dict_ema"]
```

Comment:
The EMA model is weird in that it never uses spectral reparametrization. In order to load this state dict, we should also do this:
```
model_args.spectral_reparam = False
```

Reply:
Fixed in latest commit thanks.